AxisRotationConstraint.cs

```diff
@@ -0,0 +1,138 @@
+using Microsoft.MixedReality.Toolkit;
+using Microsoft.MixedReality.Toolkit.SpatialManipulation;
+using UnityEngine;
+
+public class AxisRotationConstraint : TransformConstraint
+{
+    #region Properties
+
+    [SerializeField]
+    [Tooltip("Set to true, if rotation around X-Axis is possible.")]
+    private bool allowRotationOnXAxis;
+
+    [SerializeField]
+    [Tooltip("Set to true, if rotation around Y-Axis is possible.")]
+    private bool allowRotationOnYAxis;
+
+    [SerializeField]
+    [Tooltip("Set to true, if rotation around Z-Axis is possible.")]
+    private bool allowRotationOnZAxis;
+
+    [SerializeField]
+    [Tooltip("Set the size of the steps for fixed rotation")]
+    private float fixedStepSize = 1.0f;
+
+    [SerializeField]
+    [Tooltip("Enable to allow rotation only in fixed steps.")]
+    private bool bFixedStep = false;
+
+    [SerializeField]
+    [Tooltip("Check if object rotation should be in local space of object being manipulated instead of world space.")]
+    private bool useLocalSpaceForConstraint = false;
+
+    /// <summary>
+    /// Gets or sets whether the constraints should be applied in local space of the object being manipulated or world space.
+    /// </summary>
+    public bool UseLocalSpaceForConstraint
+    {
+        get => useLocalSpaceForConstraint;
+        set => useLocalSpaceForConstraint = value;
+    }
+
+    public override TransformFlags ConstraintType => TransformFlags.Rotate;
+
+    #endregion Properties
+
+
+    private void Awake()
+    {
+        allowRotationOnXAxis = true;
+        allowRotationOnYAxis = true;
+        allowRotationOnZAxis = true;
+    }
+
+
+    #region Public Methods
+
+    /// <summary>
+    /// Removes rotation about given axis if its flag is found
+    /// in ConstraintOnRotation
+    /// </summary>
+    public override void ApplyConstraint(ref MixedRealityTransform transform)
+    {
+        Quaternion rotation = transform.Rotation * Quaternion.Inverse(WorldPoseOnManipulationStart.Rotation);
+        Vector3 eulers = rotation.eulerAngles;
+
+        if (bFixedStep)
+        {
+            Vector3 eulerStart = Quaternion.Inverse(WorldPoseOnManipulationStart.Rotation).eulerAngles;
+
+            eulers.x = ((int)(eulers.x / fixedStepSize)) * fixedStepSize;
+            eulers.y = ((int)(eulers.y / fixedStepSize)) * fixedStepSize;
+            eulers.z = ((int)(eulers.z / fixedStepSize)) * fixedStepSize;
+        }
+
+
+        if (!allowRotationOnXAxis)

```

Comment:
style: please avoid if statements with no brackets.  (style guide is pending)

change all if statements to 

```
if (...)
{
   ....
}
```

if (